master/internal/rm/agentrm/agent.go

```diff
@@ -501,17 +536,12 @@ func (a *agent) handleIncomingWSMessage(ctx *actor.Context, msg aproto.MasterMes
 			if err != nil {
 				log.WithError(err).
 					Error("change in agent devices was detected")
-				wsm := ws.WriteMessage{
-					Message: aproto.AgentMessage{
-						AgentShutdown: &aproto.AgentShutdown{
-							ErrMsg: aproto.ErrAgentMustReconnect.Error(),
-						},
+				a.socket.Outbox <- aproto.AgentMessage{
+					AgentShutdown: &aproto.AgentShutdown{
+						ErrMsg: aproto.ErrAgentMustReconnect.Error(),
 					},
 				}
-				if err = ctx.Ask(a.socket, wsm).Error(); err != nil {
-					log.WithError(err).Error("failed to tell agent to reconnect")
-					panic(err)
-				}
+
 				ctx.Self().Stop()
```

Comment:
note, actor system stops your children when you stop. i think we're covered by the ws close in poststop, just pointing it out.

Reply:
Ah, yes, good point.